packages/gatsby/src/utils/get-react-hot-loader-strategy.ts

```diff
@@ -7,10 +7,13 @@ export function getReactHotLoaderStrategy(): string {
   // If the user has defined this, we don't want to do any package sniffing
   if (process.env.GATSBY_HOT_LOADER) return process.env.GATSBY_HOT_LOADER
 
+  // If the config flag is true, return fast-refresh
+  if (process.env.GATSBY_FAST_REFRESH) return `fast-refresh`
```

Comment:
This order of conditions has confusing impact on terminal output when user adds `FAST_REFRESH` to config flags, but still use `GATSBY_HOT_LOADER=react-hot-loader`:

```
➜  gatsby-starter-blog git:(master) ✗ GATSBY_HOT_LOADER=react-hot-lodear gatsby develop
info The following flags are active:
- FAST_REFRESH · (Umbrella Issue (​https://github.com/gatsbyjs/gatsby/discussions/28390​)) · Use React Fast Refresh instead of the legacy react-hot-loader for instantaneous feedback in your development
 server. Recommended for versions of React >= 17.0.
```

but in the end `react-hot-loader` will be used because this code as-is prefers it

It's tricky because it feels like env vars should have priority over config (same as cli toggles over any config in file). So the order now seems correct - just the message is confusing :/

Reply:
maybe before we handle flags we need to do some hacky stuff like change https://github.com/gatsbyjs/gatsby/blob/7577e3ffc1c3ac31554311014b0fa0dd8b78ebb9/packages/gatsby/src/services/initialize.ts#L186-L187

to be something like
```js
  if (config && config.flags) {
    if(config.flags.?FAST_REFRESH && process.env.GATSBY_HOT_LOADER) {
      delete config.flags.FAST_REFRESH
      reporter.warn(`Both FAST_REFRESH config flag and GATSBY_HOT_LOADER env var is used. Will use env var`)
    }
    const availableFlags = require(`../utils/flags`).default
```

(:

```
if(config.flags.?FAST_REFRESH && process.env.GATSBY_HOT_LOADER) {

}
```

Reply:
I like hacky! Well temporarily hacky that we have nice TODOs about fixing but that unblocks us to ship.

Reply:
Pushed https://github.com/gatsbyjs/gatsby/pull/28409/commits/c803d2b5f9c9d9600268458dbe9df6c0eafb0e0f - 
![Screenshot 2020-12-01 at 18 06 43](https://user-images.githubusercontent.com/419821/100772567-0e549280-3400-11eb-8cd7-48c405e3232c.png)